client/v2/algod/algod.go

```diff
@@ -0,0 +1,88 @@
+package algod
+
+import (
+	"context"
+	"github.com/algorand/go-algorand-sdk/client/v2/common"
+)
+
+type Client common.Client
+
+// get performs a GET request to the specific path against the server, assumes JSON response
+func (c *Client) get(ctx context.Context, response interface{}, path string, request interface{}, headers []*common.Header) error {
+	return (*common.Client)(c).Get(ctx, response, path, request, headers)
+}
+
+// getMsgpack performs a GET request to the specific path against the server, assumes msgpack response
+func (c *Client) getMsgpack(ctx context.Context, response interface{}, path string, request interface{}, headers []*common.Header) error {
```

Comment:
Note that if there was an error, the error response will be JSON. You can check that the status code is 200 to decide whether to treat it as a messagepack object.

Reply:
:+1: the msgpack decoder is only used in the non-error path